### PR TITLE
Add documentation and release notes for regexp/joni

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

File: presto-docs/src/main/sphinx/functions/regexp.rst [hunk 1]
```diff
@@ -2,7 +2,21 @@
...
+* When using multi-line mode (enabled via the ``(?m)`` flag),
+  only ``\n`` is recognized as a line terminator.
+* Case-insensitive matching (enabled via the ``(?i)`` flag) is always
+  performed in a Unicode-aware way. However, context-sensitive and
+  local-sensitive matching is not supported. Additionally, the
+  ``(?u)`` flag is not supported and must not be used.
+* ``\uD800\uDC00`` are not treated as ``U+10000``. Instead, use ``\x{10000}``.
+* ``\b`` incorrectly handles a non-spacing mark without base characters.
+* ``\Q`` and ``\E`` are not supported in character classes (``[A-Z123]``).
+* For Unicode character class, prefixes including ``Is``, ``blk=``, ``block=``,
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/regexp.rst]:
```
* Unicode character classes (``\p{prop}``) are supported with
  the following differences:
  * Scripts must be specified directly, without the
    ``Is``, ``script=`` or ``sc=`` prefixes.
    Example: ``\p{Hiragana}``
  * Blocks must be specified with the ``In`` prefix.
    The ``block=`` and ``blk=`` prefixes are not supported.
    Example: ``\p{Mongolian}``
  * Categories must be specified directly, without the ``Is``,
    ``general_category=`` or ``gc=`` prefixes.
    Example: ``\p{L}``

**Comment by haozhun** [comment]
Please review again.

File: presto-docs/src/main/sphinx/functions/regexp.rst [hunk 2]
```diff
@@ -56,4 +83,6 @@ All of the regular expression functions use the `Java pattern`_ syntax.
 
 .. function:: regexp_split(string, pattern) -> array<varchar>
 
-    Splits ``string`` using the regular expression ``pattern`` and returns an array.
\ No newline at end of file
+    Splits ``string`` using the regular expression ``pattern`` and returns an
+    array. Unlike in Java, trailing empty strings are not removed. They are
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/regexp.rst]: Replace the last two sentences with "Trailing empty strings are preserved."

File: presto-docs/src/main/sphinx/release/release-0.102.rst [hunk 3]
```diff
@@ -15,6 +15,13 @@ Additionally, the functions operate on Unicode code points and not user visible
 into a single user-perceived *character*, the basic unit of a writing system for a
 language, but the functions will treat each code point as a separate unit.
 
+Regular Expression
+------------------
+
+All regular expression functions have been rewritten to address performance
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.102.rst]: "rewritten to improve performance."

Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.102.rst]:
```
All :doc:`/functions/regexp` have been rewritten to improve performance.
The new versions are often twice as fast and in some cases can be many
orders of magnitude faster (due to removal of asymptotic behavior).
This change introduced some minor incompatibilities that are explained
in the function documentation.

**Pull request merged**